azext_edge/edge/commands_namespaces.py

```diff
@@ -917,3 +918,549 @@ def query_namespace_assets(
         device_name=device_name,
         device_endpoint_name=device_endpoint_name
     )
+
+
+# ASSET DATASET COMMANDS
+def add_namespace_custom_asset_dataset(
+    cmd,
+    asset_name: str,
+    namespace_name: str,
+    resource_group_name: str,
+    dataset_name: str,
+    dataset_data_source: str,
+    dataset_configuration: Optional[str] = None,
+    dataset_destinations: Optional[str] = None,
+    replace: Optional[bool] = False,
+    **kwargs
+) -> dict:
+    return NamespaceAssets(cmd).add_dataset(
+        asset_name=asset_name,
+        namespace_name=namespace_name,
+        resource_group_name=resource_group_name,
+        dataset_name=dataset_name,
+        asset_type="custom",
+        dataset_data_source=dataset_data_source,
+        datasets_custom_configuration=dataset_configuration,
+        dataset_destinations=dataset_destinations,
+        replace=replace,
+        **kwargs
+    )
+
+
+def add_namespace_opcua_asset_dataset(
+    cmd,
+    asset_name: str,
+    namespace_name: str,
+    resource_group_name: str,
+    dataset_name: str,
+    dataset_data_source: str,
+    dataset_destinations: Optional[str] = None,
+    opcua_dataset_publishing_interval: Optional[int] = None,
+    opcua_dataset_sampling_interval: Optional[int] = None,
+    opcua_dataset_queue_size: Optional[int] = None,
+    opcua_dataset_key_frame_count: Optional[int] = None,
+    opcua_dataset_start_instance: Optional[str] = None,
+    replace: Optional[bool] = False,
+    **kwargs
+) -> dict:
+    return NamespaceAssets(cmd).add_dataset(
+        asset_name=asset_name,
+        namespace_name=namespace_name,
+        resource_group_name=resource_group_name,
+        dataset_name=dataset_name,
+        asset_type=DeviceEndpointType.OPCUA.value,
```

Comment:
use constant here but "custom" string above?

Reply:
asset type is a place holder to show which "asset type" is being used (we don't want customers to have a media device endpoint and then make an asset with opcua configs in most situations)

Custom is a place holder for "do whatever you want" - useful for testing or 3rd party connectors